From d1e7a21f88eaf50298b02019b61b75e03c75bbaa Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:23:48 -0400
Subject: [tools] perf/build: split out feature check 'strlcpy'

Message-id: <1408462094-14194-60-git-send-email-jolsa@redhat.com>
Patchwork-id: 88045
O-Subject: [PATCH RHEL7.1 BZ1131394 059/325] tools/perf/build: Split out feature check: 'strlcpy'
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit d0707c9172e5d286fc119dcef5ea1621ef3bfade
Author: Ingo Molnar <mingo@kernel.org>
Date: Tue Oct 8 10:10:45 2013 +0200
---
 create mode 100644 tools/perf/config/feature-checks/test-strlcpy.c

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/config/Makefile b/tools/perf/config/Makefile
index 7e13969..c8ac4df 100644
--- a/tools/perf/config/Makefile
+++ b/tools/perf/config/Makefile
@@ -121,6 +121,7 @@ FEATURE_TESTS =    \
  libperl    \
  libpython   \
  libpython-version  \
+ strlcpy    \
  libbfd    \
  libnuma
 
@@ -434,7 +435,7 @@ else
 endif
 
 ifndef NO_STRLCPY
-  ifeq ($(call try-cc,$(SOURCE_STRLCPY),,-DHAVE_STRLCPY_SUPPORT),y)
+  ifeq ($(feature-strlcpy), 1)
     CFLAGS += -DHAVE_STRLCPY_SUPPORT
   endif
 endif
diff --git a/tools/perf/config/feature-checks/Makefile b/tools/perf/config/feature-checks/Makefile
index af65aaa..d348aa0 100644
--- a/tools/perf/config/feature-checks/Makefile
+++ b/tools/perf/config/feature-checks/Makefile
@@ -19,6 +19,7 @@ FILES=     \
  test-libperl   \
  test-libpython   \
  test-libpython-version  \
+ test-strlcpy   \
  test-libbfd   \
  test-libnuma
 
@@ -113,6 +114,9 @@ test-libpython:
 test-libpython-version:
  $(BUILD) $(FLAGS_PYTHON_EMBED)
 
+test-strlcpy:
+ $(BUILD)
+
 test-libbfd:
  $(BUILD) -DPACKAGE='perf' -DPACKAGE=perf -lbfd -ldl
 
diff --git a/tools/perf/config/feature-checks/test-strlcpy.c b/tools/perf/config/feature-checks/test-strlcpy.c
new file mode 100644
index 0000000..4a6b6ff
--- /dev/null
+++ b/tools/perf/config/feature-checks/test-strlcpy.c
@@ -0,0 +1,8 @@
+#include <stdlib.h>
+extern size_t strlcpy(char *dest, const char *src, size_t size);
+
+int main(void)
+{
+ strlcpy(NULL, NULL, 0);
+ return 0;
+}
-- 
1.7.1